pkgs/development/zig-packages/build-zig-package/default.nix

```diff
@@ -0,0 +1,39 @@
+{ lib, stdenv, zig }:
+
+{ strictDeps ? true
+, nativeBuildInputs ? [ ]
+, enableParallelBuilding ? true
+, meta ? { }
+, ...
+}@attrs:
+
+stdenv.mkDerivation (attrs // {
+  inherit strictDeps enableParallelBuilding;
+  nativeBuildInputs = [ zig ] ++ nativeBuildInputs;
+
+  postConfigure = ''
+    export HOME=$TMPDIR
+  '';
+
+  buildPhase = attrs.buildPhase or ''
+    runHook preBuild
+    zig build -Drelease-safe -Dcpu=baseline
+    runHook postBuild
+  '';
+
+  checkPhase = attrs.checkPhase or ''
+    runHook preCheck
+    zig build test
+    runHook postCheck
+  '';
+
+  installPhase = attrs.installPhase or ''
+    runHook preInstall
+    zig build -Drelease-safe -Dcpu=baseline --prefix $out install
```

Comment:
Here I would like to hear the opinion of @andrewrk as lang creator:

Usually in autotools software, the command to install is just `make install`, whereas  here it is `zig build $buildFlags install`.
Is this expected? 
To me it looks like the Zig compiler will re-execute the build logic two times, one in buildPhase and again in installPhase.

Reply:
It doesn't rebuild twice, this was the only way to copy the result to the nix store without throwing an access denied error